sycl/include/sycl/ext/oneapi/experimental/graph.hpp

```diff
@@ -168,6 +168,24 @@ class __SYCL_EXPORT modifiable_command_graph {
   /// executing.
   bool end_recording(const std::vector<queue> &RecordingQueues);
 
+  /// Prints contents of graph to text file in DOT format
+  /// @param GraphName is a string appended to the output file name
+  void printGraphAsDot(const char *GraphName);
+
+  /// Checks is the graph_impl of Graph has a similar structure to
+  /// the graph_impl of the caller.
+  /// Graphs are considered similar if they have same numbers of nodes
+  /// of the same type with similar predecessor and successor nodes (number and
+  /// type). Two nodes are considered similar if they have the same
+  /// command-group type. For command-groups of type "kernel", the "signature"
+  /// of the kernel is also compared (i.e. the name of the command-group).
+  /// @param Graph if reference to the graph to compare with.
+  /// @param DebugPrint if set to true print additional debug information about
+  /// the spotted graph differences
+  /// @return true if the two graphs are similar, false otherwise
+  bool hasSimilarStructure(const command_graph<graph_state::modifiable> &Graph,
+                           bool DebugPrint = false) const;
+
```

Comment:
This `graph.hpp` file is user facing, so the camelCase LLVM convention for function names doesn't apply, it should be snake_case like the rest of the SYCL spec.

We've also not specified these new functions in our extensions spec, so we'd need to discuss what our expectations are if users start using this. As an experimental extension, we're allowed to have API breakages as we iterate, but we still shouldn't modify the public API without some consideration/plan.

If these new functions are just for testing right now, can we use unittests to get a `graph_impl` object and interface with the methods at that level?